android-record/src/main/java/com/fasterxml/jackson/module/androidrecord/AndroidRecordModule.java

```diff
@@ -0,0 +1,167 @@
+ *   Android Developers Blog article</a>
+ *
+ * <p>
+ * Note: this module is a no-op when no Android-desugared records are being deserialized,
+ * so it is safe to use in code shared between Android and non-Android platforms.
+ *
+ * <p>
+ * Note: the canonical record constructor is found
+ * through matching of parameter names and types with fields.
+ * Therefore, this module doesn't allow a deserialized desugared record class to have a custom
+ * constructor with the same set of parameter names and types as the canonical one.
+ *
+ * @author Eran Leshem
+ **/
+public class AndroidRecordModule extends SimpleModule {
+  private static final class AndroidRecordNaming
+      extends DefaultAccessorNamingStrategy
+  {
+      /**
+       * Names of actual Record components from definition; auto-detected.
+       */
+      private final Set<String> _componentNames;
+
+      private AndroidRecordNaming(MapperConfig<?> config, AnnotatedClass forClass) {
+          super(config, forClass,
+                  // no setters for (immutable) Records:
+                  null,
+                  // trickier: regular fields are ok (handled differently), but should
+                  // we also allow getter discovery? For now let's do so
+                  "get", "is", null);
+          _componentNames = getDesugaredRecordComponents(forClass.getRawType()).map(Field::getName)
+                  .collect(Collectors.toSet());
+      }
+
+      @Override
+      public String findNameForRegularGetter(AnnotatedMethod am, String name)
+      {
+          // By default, field names are un-prefixed, but verify so that we will not
+          // include "toString()" or additional custom methods (unless latter are
+          // annotated for inclusion)
+          if (_componentNames.contains(name)) {
+              return name;
+          }
+          // but also allow auto-detecting additional getters, if any?
+          return super.findNameForRegularGetter(am, name);
+      }
+  }
+
+  private static class AndroidRecordClassIntrospector extends BasicClassIntrospector {
+    @Override
+    protected POJOPropertiesCollector collectProperties(MapperConfig<?> config, JavaType type, MixInResolver r,
+                                                        boolean forSerialization) {
+      if (isDesugaredRecordClass(type.getRawClass())) {
+        AnnotatedClass classDef = _resolveAnnotatedClass(config, type, r);
+        AccessorNamingStrategy accNaming = new AndroidRecordNaming(config, classDef);
+        return constructPropertyCollector(config, classDef, type, forSerialization, accNaming);
+      }
+
+      return super.collectProperties(config, type, r, forSerialization);
+    }
+  }
+
+  @Override
+  public void setupModule(SetupContext context) {
+    super.setupModule(context);
+    context.addValueInstantiators(AndroidRecordModule::findValueInstantiator);
+    context.setClassIntrospector(new AndroidRecordClassIntrospector());
```

Comment:
Hmmmmh. This is somewhat problematic as it does not compose; ideally Modules wouldn't have to override it, as only one module can really change it (others would override it).
I'll need to think about this a bit.

Reply:
Does that mean that my statement in the doc that "this module is a no-op when no Android-desugared records are being (de)serialized" is not accurate anymore? That *is* problematic.

Reply:
Only if some other module tries to override `ClassIntrospector` -- it is not problem if this is the only module to do so.
So unlike, say, `AnnotationIntrospector`s that can be chained, there's no existing mechanism to combine `ClassIntrospector`s

Reply:
How common is it for modules to do that? Should I document this?

Reply:
@eranl It might be best to add something in README to mention this.
I don't remember any other module overriding it, so this is theoretical problem, not actual one.
(at least wrt modules under FasterXML)

On chaining... I thought about that, but in general I'm not sure how commonly this is needed etc.
Due to timing it needs to wait until 2.17 (that is, post 2.16.0 and cannot go in a patch) so there's no hurry.
So what I am thinking is that for 2.17 we could consider various approaches to remove this problem; and not have to make last minute changes that may turn out problematic.

Reply:
> @eranl It might be best to add something in README to mention this. I don't remember any other module overriding it, so this is theoretical problem, not actual one. (at least wrt modules under FasterXML)

Done. see #230.

> On chaining... I thought about that, but in general I'm not sure how commonly this is needed etc. Due to timing it needs to wait until 2.17 (that is, post 2.16.0 and cannot go in a patch) so there's no hurry. So what I am thinking is that for 2.17 we could consider various approaches to remove this problem; and not have to make last minute changes that may turn out problematic.

Sure, I meant in a future version. Please let me know when is a good time, and what's your preferred approach.